include/utransport/UMessageType.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ #ifndef _UMESSAGETYPE_H_
+ #define _UMESSAGETYPE_H_
+
+#include <iostream>
+#include <optional>
+
+namespace uprotocol 
+{
+    namespace utransport
+    {
+        namespace datamodel
+        {
+            enum class UMessageType {
+                PUBLISH = 0,    // Publish or notification event
+                REQUEST = 1,    // Request
+                RESPONSE = 2,   // Response
+                UNDEFINED = 3
```

Comment:
Explicit enum value allocation

Reply:
Can we avoid to enumerate everything, less is more.?